pandas/core/indexing.py

```diff
@@ -927,28 +928,8 @@ def _multi_take(self, tup):
         """
         try:
             o = self.obj
-            d = {}
-            for key, axis in zip(tup, o._AXIS_ORDERS):
-                ax = o._get_axis(axis)
-                # Have the index compute an indexer or return None
-                # if it cannot handle:
-                indexer, keyarr = ax._convert_listlike_indexer(key,
-                                                               kind=self.name)
-                # We only act on all found values:
-                if indexer is not None and (indexer != -1).all():
-                    self._validate_read_indexer(key, indexer, axis)
-                    d[axis] = (ax[indexer], indexer)
-                    continue
-
-                # If we are trying to get actual keys from empty Series, we
-                # patiently wait for a KeyError later on - otherwise, convert
-                if len(ax) or not len(key):
-                    key = self._convert_for_reindex(key, axis)
-                indexer = ax.get_indexer_for(key)
-                keyarr = ax.reindex(keyarr)[0]
-                self._validate_read_indexer(keyarr, indexer,
-                                            o._get_axis_number(axis))
-                d[axis] = (keyarr, indexer)
+            d = {axis: self._get_listlike_indexer(key, axis)
+                 for (key, axis) in zip(tup, o._AXIS_ORDERS)}
             return o._reindex_with_indexers(d, copy=True, allow_dups=True)
         except (KeyError, IndexingError) as detail:
```

Comment:
This can *only* raise KeyError now, right?

Reply:
and in fact, maybe can just use ``self._exception()`` in ``_get_listlike_indexer``? then can simply remove the try/except (here)? (and I think you do this below)

Reply:
In fact, I think the entire block was just superfluous: if we get here, it means we have collections of keys, and if a key is not found, it will raise later.